### PR TITLE
Microverse: LeaderBoard: Using Api

### PR DESCRIPTION
**Pull request by Shubh-Dev**
Hello Dear reviewer,
This is the second part of the project. In this step, I used API to post and get data.
Let me know if any corrections are required.

### CODE REVIEW

File: modules/display.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export const url = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/zRPcLNnykoeas7KKzkpU/scores';
+
+export const mainDynamicBox = document.querySelector('.dynamic-box');
+// const gameID = 'zRPcLNnykoeas7KKzkpU';
+
+export default async function displayData() {
```
Comment (MahmoudBakr23) [inline modules/display.js]: - [ ] Please convert this function to using the arrow function instead in order to practice the new ES6 features. 😄 🙏

File: modules/display.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export const url = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/zRPcLNnykoeas7KKzkpU/scores';
+
+export const mainDynamicBox = document.querySelector('.dynamic-box');
+// const gameID = 'zRPcLNnykoeas7KKzkpU';
```
Comment (MahmoudBakr23) [inline modules/display.js]: - [ ] Kindly remove any commented-out lines for a cleaner code base. 🙏

File: modules/display.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+export default async function displayData() {
+  const fetchData = await fetch(url).then((response) => response.json());
+  fetchData.result.sort((a, b) => b.score - a.score);
+  fetchData.result.forEach((item) => {
+    const dynamicDiv = document.createElement('div');
+    dynamicDiv.className = 'dynamic-div';
+    dynamicDiv.innerHTML = `${item.user} : ${item.score}`;
+    mainDynamicBox.appendChild(dynamicDiv);
+    mainDynamicBox.style.display = 'block';
```
Comment (MahmoudBakr23) [inline modules/display.js:13]: **OPTIONAL**
- You can break down this function into separate functions in order to make the logic aside from the DOM manipulation code. This helps with the performance and readiness of your app. 👍

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @Shubh-Dev  😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Descriptive README file and PR
- Well done implementing the POST request to the API

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
_Please check the comments under the review._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
### Hi @Shubh-Dev, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**